### PR TITLE
Add note that page token must be validated

### PR DESCRIPTION
**Pull request by PeterNovotney**
The page token should not be treated as trusted input. It should be validated and any access to resources specified in the page token should be authenticated. That is, the page token should not be trusted as proving an earlier authentication occurred.

### CODE REVIEW

File: aip/general/0158.md [hunk 1]
```diff
@@ -103,6 +103,10 @@ contain sensitive data, an API **may** obfuscate the page token by defining an
 internal protocol buffer message with any data needed, and send the serialized
 proto, base-64 encoded.
 
+Even with obfuscation protection, the page token might still be modified by a
+caller. The service **must** validate the contents of the page token and authenticate
+the caller for any access to resources specified in the page token.
+
```
Comment (jgeewax) [inline aip/general/0158.md:109]: This is interesting because it implies the page token includes some level of access control (like an OAuth2 access token) -- which it should not. 

Perhaps we can rephrase this to clarify that the page token must be useful only as a cursor and should carry with it no implication of any form of authorization to list or retrieve any of the items involved in a paginated list response. 

```suggestion
Page tokens **must** be limited to providing an indication of where to continue
the pagination process only. They **must not** provide any form of authorization
to the underlying resources, and authorization must be performed on the request as
with any other regardless of the presence of a page token.
```

Comment (PeterNovotney) [reply aip/general/0158.md]: Thanks, I think that makes it more clear. The main issue I was getting at was that if the page token contains some resource reference, the service should not blindly use that without going through proper checks. Your description makes the advice more direct to that goal.

File: aip/general/0158.md [hunk 2]
```diff
@@ -103,6 +103,10 @@ contain sensitive data, an API **may** obfuscate the page token by defining an
 internal protocol buffer message with any data needed, and send the serialized
 proto, base-64 encoded.
 
+Page tokens **must** be limited to providing an indication of where to continue
+the pagination process only. They **must not** provide any form of authorization
+to the underlying resources, and authorization must be performed on the request as
+with any other regardless of the presence of a page token.
 ### Expiring page tokens
```
Comment (lukesneeringer) [inline aip/general/0158.md:110]:
```suggestion
with any other regardless of the presence of a page token.

### Expiring page tokens
```

File: aip/general/0158.md [hunk 3]
```diff
@@ -103,6 +103,10 @@ contain sensitive data, an API **may** obfuscate the page token by defining an
 internal protocol buffer message with any data needed, and send the serialized
 proto, base-64 encoded.
 
+Page tokens **must** be limited to providing an indication of where to continue
+the pagination process only. They **must not** provide any form of authorization
+to the underlying resources, and authorization must be performed on the request as
```
Comment (jgeewax) [inline aip/general/0158.md:108]:
```suggestion
to the underlying resources, and authorization **must be** performed on the request as
```

File: aip/general/0158.md [hunk 4]
```diff
@@ -103,6 +103,10 @@ contain sensitive data, an API **may** obfuscate the page token by defining an
...
+with any other regardless of the presence of a page token.
```
Comment (jgeewax) [inline aip/general/0158.md:109]:
```suggestion
with any other regardless of the presence of a page token.

```

File: aip/general/0158.md (hunk 2 above)
Comment (lukesneeringer) [inline aip/general/0158.md:110]:
```suggestion
to the underlying resources, and authorization **must** be performed on the request
as with any other regardless of the presence of a page token.

### Expiring page tokens
```
